pkg/registry/file/vulnsummarystorage.go

```diff
@@ -0,0 +1,240 @@
+	return storage.NewMethodNotImplementError(key, "")
+}
+
+// add uni test
+func (s *VulnSummaryStorageImpl) Watch(ctx context.Context, key string, _ storage.ListOptions) (watch.Interface, error) {
+	_, span := otel.Tracer("").Start(ctx, "VulnSummaryStorageImpl.Watch")
+	span.SetAttributes(attribute.String("key", key))
+	defer span.End()
+	return nil, storage.NewMethodNotImplementError(key, "")
+}
+
+// add uni test
+func (s *VulnSummaryStorageImpl) getOneVulnManifestSummary(ctx context.Context, path string) (*softwarecomposition.VulnerabilityManifestSummary, error) {
+	var vulnManifestSummary softwarecomposition.VulnerabilityManifestSummary
+
+	b, err := afero.ReadFile(s.appFs, path)
+	if err != nil {
+		if errors.Is(err, afero.ErrFileNotFound) {
+			return nil, storage.NewKeyNotFoundError(path, 0)
+		}
+		logger.L().Ctx(ctx).Error("read file failed", helpers.Error(err), helpers.String("key", path))
+		return nil, err
+	}
+	err = json.Unmarshal(b, &vulnManifestSummary)
+	if err != nil {
+		logger.L().Ctx(ctx).Error("json unmarshal failed", helpers.Error(err), helpers.String("key", path))
+		return nil, err
+	}
+	if _, exist := vulnManifestSummary.GetLabels()["kubescape.io/workload-namespace"]; !exist {
+		return nil, storage.NewKeyNotFoundError(path, 0)
+	}
+	return &vulnManifestSummary, nil
+}
+
+// add uni test
+func initNamespaceVulnSummary(key string) softwarecomposition.VulnerabilitySummary {
+	namespace := filepath.Base(key)
+	var vulnSummaryByNamespace softwarecomposition.VulnerabilitySummary
+	vulnSummaryByNamespace.APIVersion = "spdx.softwarecomposition.kubescape.io/v1beta1"
+	vulnSummaryByNamespace.Kind = "VulnerabilitySummary"
+	vulnSummaryByNamespace.Labels = map[string]string{"kubescape.io/workload-namespace": namespace}
+	vulnSummaryByNamespace.Annotations = map[string]string{"kubescape.io/status": ""}
+	vulnSummaryByNamespace.Name = namespace
+	vulnSummaryByNamespace.Namespace = namespace
+	vulnSummaryByNamespace.CreationTimestamp.Time = time.Now()
+
+	return vulnSummaryByNamespace
+}
+
+func updateVulnCounters(aggregatedCounters *softwarecomposition.VulnerabilityCounters, counters *softwarecomposition.VulnerabilityCounters) {
+	aggregatedCounters.All += counters.All
+	aggregatedCounters.Relevant += counters.Relevant
+}
+
+func updateSeverities(aggregatedSeverities *softwarecomposition.SeveritySummary, severities *softwarecomposition.SeveritySummary) {
+	updateVulnCounters(&aggregatedSeverities.Critical, &severities.Critical)
+	updateVulnCounters(&aggregatedSeverities.High, &severities.High)
+	updateVulnCounters(&aggregatedSeverities.Medium, &severities.Medium)
+	updateVulnCounters(&aggregatedSeverities.Low, &severities.Low)
+	updateVulnCounters(&aggregatedSeverities.Negligible, &severities.Negligible)
+	updateVulnCounters(&aggregatedSeverities.Unknown, &severities.Unknown)
+}
+
+func updateFullVulnSumm(fullVulnSumm *softwarecomposition.VulnerabilitySummary, vulnManifestSumm *softwarecomposition.VulnerabilityManifestSummary) {
```

Comment:
thought: should these be struct methods for the appropriate structs?

Seeing all these functions take an API object of some kind made me wonder if they should be methods on corresponding structs. For example, the `updateFullVulnSumm` with its terse name seem like a `VulnerabilitySummary.Merge(other)` method to me. Same for the updating the vulnerability counters. Wouldn’t they make more sense as `VulnerabilityCounters.Add(other)` method?